src/RFC-0313_VNRegistration.md

```diff
@@ -0,0 +1,344 @@
+2. Each registration expires after a number of epochs, at which point the validator may no longer participate in [DAN] consensus.
+3. A validator MAY re-register before or after the expiration epoch is reached to allow continued participation in [DAN] consensus.
+4. A validator registration MUST be submitted as a base layer `ValidatorNodeRegistration` [UTXO] signed by the `VN_Public_Key`
+5. A validator MUST be assigned a deterministic but randomized `VN_Shard_Key` that can be verified at any epoch by other validators by inspecting the base layer.
+6. The `VN_Shard_Key` MUST be periodically reassigned/shuffled to prevent prolonged control over a particular [shard space].
+7. A validator MUST be able to generate a Merkle proof that it is registered for the epoch that it is currently participating.
+8. The validator set for any given epoch MUST be unambiguous between validators or anyone observing the [base layer] chain.
+9. Base layer reorgs MUST NOT negatively affect the [DAN] layer. 
+
+We define the following validator variables:
+
+| Symbol       | Name            | Description                                        |
+|:-------------|:----------------|:---------------------------------------------------|
+| $P_i$        | `VN_Public_Key` | The $i$th public validator node key                |
+| $S_i$        | `VN_Shard_Key`  | The $i$th 256-bit VN shard key.                    |
+| $\epsilon_i$ | `Epoch`         | The $i$th epoch. An epoch is `EpochLength` blocks. |
+
+An epoch $\epsilon$ is defined by the [base layer] block height $h$, where $\epsilon_i = \lfloor \frac{h}{\text{EpochLength}} \rfloor$,
+and spans all blocks from the start of the epoch up to but excluding the start of the next epoch.
+
+*Base layer consensus constants* (all values TBD):
+
+| Name                           | Value              | Description                                                                                |
+|:-------------------------------|:-------------------|:-------------------------------------------------------------------------------------------|
+| `EpochLength`                  | 60 Blocks (~2h)    | The number of blocks in an epoch                                                           |
+| `VNRegistrationValidityPeriod` | 20 Epochs (~40hrs) | The number of epochs that a validator node registration is valid                           |
+| `VNRegDepositAmount`           | TBD Tari           | The minimum amount that must be spent to create a valid `ValidatorNodeRegistration` [UTXO] |
+| `VNRegLockHeight`              | 10 Epochs          | The lock height that must be set on every `ValidatorNodeRegistration` [UTXO]               |
+| `VNShardShuffleInterval`       | 100 Epochs         | The interval that a validator node shard key is shuffled                                   |
+| `VNRegInstateDelay`            | 2 Epochs           | The number of epochs to delay for before instating a new VN                                |
```

Comment:
Suggestion: Activation instead of instate
```suggestion
| `VNRegActivationDelay`            | 2 Epochs           | The number of epochs to delay for before activating a new VN                                |
```